File: src/main/java/com/heekng/celloct/entity/WorkUpdateRequest.java
package com.heekng.celloct.entity;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

import static javax.persistence.FetchType.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class WorkUpdateRequest extends BaseTimeEntity {

    @Id @GeneratedValue
    @Column(name = "work_update_request_id")
    private Long id;

    @Embedded
    WorkTime workTime;

    @Column(name = "note")
    String note;

    @OneToOne(fetch = LAZY)
    @JoinColumn(name = "work_id", nullable = false, unique = true)
    private Work work;

    @Builder
    public WorkUpdateRequest(LocalDate workDate, LocalDateTime startDate, LocalDateTime endDate, Work work, String note) {
        this.work = work;
        this.workTime = WorkTime.builder()
                .workDate(workDate)
                .startDate(startDate)
                .endDate(endDate)
                .build();
        this.note = note;
    }
}

File: src/main/kotlin/com/heekng/celloct/entity/WorkUpdateRequest.kt
package com.heekng.celloct.entity

import java.time.LocalDate
import java.time.LocalDateTime
import javax.persistence.*

@Entity
class WorkUpdateRequest(
    @Id @GeneratedValue @Column(name = "work_update_request_id")
    val id: Long? = null,

    @Embedded
    val workTime: WorkTime,

    @Column(name = "note")
    val note: String? = null,

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "work_id", nullable = false, unique = true)
    val work: Work,
) : BaseTimeEntity() {

    companion object {
        fun fixture(
            workDate: LocalDate,
            startDate: LocalDateTime,
            endDate: LocalDateTime,
            work: Work,
            note: String? = null
        ): WorkUpdateRequest {
            return WorkUpdateRequest(
                work = work,
                workTime = WorkTime(
                    workDate = workDate,
                    startDate = startDate,
                    endDate = endDate,
                ),
                note = note
            )
        }
    }

}